### PR TITLE
Feature: disallow_robots task

### LINKED ISSUE #3: robots.txt
I have this task to make sure that staging doesn't get indexed.
Should we include it into bender? I use it on every project...

``` rb
namespace :deploy do
  desc "Disallow robots unless deploying to production."
  task :disallow_robots do
    return if fetch(:stage) == :production

    on roles :app do |host|
      as fetch(:application) do
        within File.join(fetch(:deploy_to), 'public') do
          execute :echo, %{"User-agent: *\nDisallow: /\n" > robots.txt}
        end
      end
    end
  end
end

after "deploy:updating", "deploy:disallow_robots"
```

### PR DESCRIPTION
**Pull request by fsmanuel**
@halo you can use the new github feature and squash the commits.

closes #3

### CODE REVIEW

File: lib/bender/capistrano/tasks/robots.rake [hunk 1]
```diff
@@ -0,0 +1,16 @@
+namespace :deploy do
+  desc "Disallow robots unless deploying to production."
+  task :disallow_robots do
+    return if fetch(:stage) == :production
```
Comment (halo) [inline lib/bender/capistrano/tasks/robots.rake]: This is a rake task block, not a method, could it be that this `return` aborts everything? Or does it actually work as intended?

File: lib/bender/capistrano/tasks/robots.rake [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      as fetch(:application) do
+        within File.join(fetch(:deploy_to), 'public') do
+          execute :echo, %{"User-agent: *\nDisallow: /\n" > robots.txt}
+        end
+      end
+    end
+  end
+end
+
+after "deploy:updating", "deploy:disallow_robots"
```
Comment (halo) [inline lib/bender/capistrano/tasks/robots.rake]: I just realize that we don't define any `after` anywhere in bender. E.g. Rails assets and migrations are not run automatically. Should we really do it here?

File: lib/bender/capistrano/tasks/robots.rake [hunk 3]
```diff
@@ -0,0 +1,16 @@
+namespace :deploy do
```
Comment (halo) [inline lib/bender/capistrano/tasks/robots.rake]: If this is in the `deploy` namespace, why not have it in the `deploy.rake`? To be a little bit more explicit about what is run where. So you don't have to guess the order of things running by looking into every file.

File: lib/bender/capistrano/tasks/robots.rake (hunk 1 above)
Comment (fsmanuel) [inline lib/bender/capistrano/tasks/robots.rake]: Good point! It should be `next`

File: lib/bender/capistrano/tasks/robots.rake (hunk 2 above)
Comment (fsmanuel) [inline lib/bender/capistrano/tasks/robots.rake]: I'll add it to the template

File: lib/bender/capistrano/tasks/robots.rake (hunk 3 above)
Comment (fsmanuel) [inline lib/bender/capistrano/tasks/robots.rake]: 👍

**Comment by fsmanuel** [comment]
@halo updated

File: lib/bender/capistrano/tasks/deploy.rake [hunk 4]
```diff
@@ -43,6 +43,23 @@ namespace :deploy do
     end
   end
 
+  # ––––––––––––––
+  # Set robots.txt
+  # ––––––––––––––
+
+  desc 'Disallow robots unless deploying to production.'
+  task :disallow_robots do
+    next if fetch(:stage) == :production
```
Comment (halo) [inline lib/bender/capistrano/tasks/deploy.rake:52]: > However what it actually does is just to return from the current block. So you can use it with any method that takes a block - even if it has nothing to do with iteration.

Look at that, didn't know that :)

**Comment by halo** [comment]
Can I squash right from the github website? I thought I'd have to do that on the CLI...

File: lib/bender/capistrano/tasks/deploy.rake (hunk 4 above)
Comment (fsmanuel) [inline lib/bender/capistrano/tasks/deploy.rake:52]: It was news to me as well!

**Comment by fsmanuel** [comment]
You should have two options now. One is the old merge and one is a squash. Go to the repo settings and make it the only option so it's the default from now on.

**Comment by halo** [comment]
Cool, I thought that setting was only preventing regular merges. I didn't know it actually effects the way the "merge" button on the website works! :) Do you want me to release a new tag?

PS: Let's introduce a `CHANGELOG.md` by the way. I suggest the same formatting as the rails gem.

**Comment by fsmanuel** [comment]
New tag please! Can we switch to SemVer so the new release would be 0.1.0?
Can you start the `CHANGELOG.md` so I can pick up the formatting from your examples?

**Pull request merged**